### ISSUE TITLE
lines-around-directive has false positive with semicolon-less style.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#7450]
**Tell us about your environment**
- **ESLint Version:** master (6e9ff08cf8ac9188331fcea7905ce162accefe81)
- **Node Version:** 6.9.1
- **npm Version:** 3.8.9

**What parser (default, Babel-ESLint, etc.) are you using?**
- default

**Please show your full configuration:**
- nothing

**What did you do? Please include the actual source code causing the issue.**

``` bash
$ eslint test.js --no-eslintrc --no-ignore
```

`test.js` is:

``` js
/* eslint lines-around-directive: error */

"use strict"

;[1, 2, 3].forEach(doSomething)
```

**What did you expect to happen?**

No errors since there is an empty line between `"use strict"` and the next statement.

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ eslint test.js --no-eslintrc --no-ignore

C:\Users\t-nagashima.AD\Documents\GitHub\eslint\test.js
  3:1  error  Expected newline after "use strict" directive  lines-around-directive

✖ 1 problem (1 error, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Related: [newline-after-var](https://github.com/eslint/eslint/blob/6e9ff08cf8ac9188331fcea7905ce162accefe81/lib/rules/newline-after-var.js#L48-L77) has logic to account for this.
